In [ ]:
%run preprocessing.ipynb 
%run model_training.ipynb

In [2]:
from sklearn.metrics import r2_score,mean_absolute_error,mean_squared_error,root_mean_squared_error

In [3]:
print("Evaluating metrics for linear regression:")

mse = mean_squared_error(y_value, y_pred_lr)
mae = mean_absolute_error(y_value, y_pred_lr)
rmse = np.sqrt(mse)
r2 = r2_score(np.expm1(y_value), y_pred_lr)

print("Mean Squared Error:", mse)
print("Mean Absolute Error:", mae)
print("Root Mean Squared Error:", rmse)
print("R squared score:", r2)

Evaluating metrics for linear regression:
Mean Squared Error: 83184832.76192433
Mean Absolute Error: 8501.487706493752
Root Mean Squared Error: 9120.571953661914
R squared score: 0.619424792455481


In [4]:
print("Evaluating metrics for Random Forest")

mse = mean_squared_error(y_value_rf_xg, y_pred_rfr)
mae = mean_absolute_error(y_value_rf_xg, y_pred_rfr)
rmse = np.sqrt(mse)
r2 = r2_score(y_value_rf_xg, y_pred_rfr)

print("Mean Squared Error:", mse)
print("Mean Absolute Error:", mae)
print("Root Mean Squared Error:", rmse)
print("R squared score:", r2)

Evaluating metrics for Random Forest
Mean Squared Error: 995568.9084062915
Mean Absolute Error: 586.5133515580645
Root Mean Squared Error: 997.7819944287888
R squared score: 0.9508769649770983


This Random Forest model is performing very well overall, as indicated by the R² score of 0.951, which means it explains about 95% of the variance in the target variable. On average, your predictions are off by around 582 units (MAE), and the typical error magnitude (RMSE) is ~993, which is higher than MAE due to a few larger errors.
This suggests that while the model is highly accurate for most data points, there are occasional larger deviations. The spread between MAE and RMSE highlights that some predictions might be outliers or extreme cases where the model underperforms.

Takeaways:
Overall, this model is reliable and fits the data well.
There may be specific instances where predictions are less accurate, so checking residuals or feature effects can help improve performance.
I could further tune hyperparameters or analyze feature importance to refine the model and possibly reduce the larger errors.

In [5]:
print("Evaluating metrics for XGBoost model:")

mse = mean_squared_error(y_value_rf_xg, y_pred_xgb)
mae = mean_absolute_error(y_value_rf_xg, y_pred_xgb)
rmse = np.sqrt(mse)
r2 = r2_score(y_value_rf_xg, y_pred_xgb)

print("Mean Squared Error:", mse)
print("Mean Absolute Error:", mae)
print("Root Mean Squared Error:", rmse)
print("R squared score:", r2)

Evaluating metrics for XGBoost model:
Mean Squared Error: 1520420.6596971932
Mean Absolute Error: 843.6956373726953
Root Mean Squared Error: 1233.0533888267746
R squared score: 0.9249799017574698
